plugins/inputs/solrmetrics/solrmetrics.go

```diff
@@ -0,0 +1,413 @@
+			"solr.jetty:org.eclipse.jetty.server.handler.DefaultHandler.head-requests:count",
+			"solr.jetty:org.eclipse.jetty.server.handler.DefaultHandler.move-requests:count",
+			"solr.jetty:org.eclipse.jetty.server.handler.DefaultHandler.delete-requests:count",
+			"solr.jetty:org.eclipse.jetty.server.handler.DefaultHandler.get-requests:count",
+			"solr.jetty:org.eclipse.jetty.server.handler.DefaultHandler.post-requests:count",
+			"solr.jetty:org.eclipse.jetty.server.handler.DefaultHandler.put-requests:count",
+			"solr.jetty:org.eclipse.jetty.server.handler.DefaultHandler.other-requests:count",
+			"solr.jvm:buffers.direct.Count",
+			"solr.jvm:buffers.direct.MemoryUsed",
+			"solr.jvm:buffers.direct.TotalCapacity",
+			"solr.jvm:buffers.mapped.Count",
+			"solr.jvm:buffers.mapped.MemoryUsed",
+			"solr.jvm:buffers.mapped.TotalCapacity",
+			"solr.jvm:threads.blocked.count",
+			"solr.jvm:threads.count",
+			"solr.jvm:threads.daemon.count",
+			"solr.jvm:threads.deadlock.count",
+			"solr.jvm:threads.new.count",
+			"solr.jvm:threads.runnable.count",
+			"solr.jvm:threads.terminated.count",
+			"solr.jvm:threads.timed_waiting.count",
+			"solr.jvm:threads.waiting.count",
+			"solr.jvm:os.maxFileDescriptorCount",
+			"solr.jvm:os.openFileDescriptorCount",
+			"solr.jvm:memory.total.init",
+			"solr.jvm:memory.total.max",
+			"solr.jvm:memory.total.used",
+			"solr.jvm:memory.heap.init",
+			"solr.jvm:memory.heap.max",
+			"solr.jvm:memory.heap.used",
+			"solr.jvm:gc.ConcurrentMarkSweep.count",
+			"solr.jvm:gc.ConcurrentMarkSweep.time",
+			"solr.jvm:gc.ParNew.count",
+			"solr.jvm:gc.ParNew.time",
+			"solr.node:CONTAINER.fs.totalSpace",
+			"solr.node:CONTAINER.fs.usableSpace",
+			"solr.node:ADMIN./admin/zookeeper.errors:count",
+			"solr.node:ADMIN./admin/zookeeper.timeouts:count",
+			"solr.node:CONTAINER.cores.lazy",
+			"solr.node:CONTAINER.cores.loaded",
+			"solr.node:CONTAINER.cores.unloaded"
+		]
+	`
+)
+
+// ResponseHeader is an exported type that
+// contains a response metrics: QTime and Status
+type ResponseHeader struct {
+	QTime  int64 `json:"QTime"`
+	Status int64 `json:"status"`
+}
+
+// Serie is type to represent series
+type Serie struct {
+	mesName   string // Measurament name
+	fields    Fields
+	tags      Tags
+	timestamp time.Time
+}
+
+// Tags for series
+type Tags map[string]string
+
+// Metrics are type to represent generic metrics returned by Solr
+type Metrics map[string]interface{}
+
+// Fields to export
+type Fields map[string]interface{}
+
+// MetricResponse is an exported type that
+// contains Metrics of different types and represents response of Solr
+type MetricResponse struct {
+	ResponseHeader ResponseHeader `json:"responseHeader"`
+	Metrics        Metrics        `json:"metrics"`
+}
+
+// Solr is a plugin to read stats from one or many Solr servers
+type Solr struct {
+	Local       bool
+	Servers     []string
+	Username    string
+	Password    string
```

Comment:
Please add toml annotations
```suggestion
	Servers     []string `toml:"servers"`
	Username    string `toml:"username"`
	Password    string `toml:"password"`
```
same for the other variables.